src/incidents/list/ViewIncidentsTable.tsx

```diff
@@ -22,33 +23,100 @@ function ViewIncidentsTable(props: Props) {
     return <Spinner type="DotLoader" loading />
   }
 
+  // filter data
+  const exportData = [{}]
+  let first = true
+  if (data != null) {
+    data.forEach((elm) => {
+      const entry = {
+        code: elm.code,
+        date: format(new Date(elm.date), 'yyyy-MM-dd hh:mm a'),
+        reportedBy: elm.reportedBy,
+        reportedOn: format(new Date(elm.reportedOn), 'yyyy-MM-dd hh:mm a'),
+        status: elm.status,
+      }
+      if (first) {
+        exportData[0] = entry
+        first = false
+      } else {
+        exportData.push(entry)
+      }
+    })
+  }
+
+  function downloadCSV() {
+    const fields = Object.keys(exportData[0])
+    const opts = { fields }
+    const parser = new Parser(opts)
+    const csv = parser.parse(exportData)
```

Comment:
Could you elaborate how to do this? Would this be in a new file?

Reply:
Yeah. It would go in a new file. 

The function would look something like

```ts
function getCSV<T>(data: T[], keys: (keyof T)[]): string {
    const opts = { fields }
    const parser = new Parser(opts)
    const csv = parser.parse(data)
    return csv
}
```ts

Then would be called like: 

```ts
getCSV(exportedData, ['code', 'date', 'reportedOn', 'status'])
```

Reply:
will give a crack at this in a day or two